x/slashing/slashing_period.go

```diff
@@ -66,6 +66,16 @@ func (k Keeper) iterateValidatorSlashingPeriods(ctx sdk.Context, handler func(sl
 	}
 }
 
+// Delete all slashing periods in the store.
+func (k Keeper) DeleteValidatorSlashingPeriods(ctx sdk.Context) {
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, ValidatorSlashingPeriodKey)
+	defer iter.Close()
```

Comment:
seems kinda silly to use defer statements in this context, but doesn't really matter

Reply:
OK, now `iter.Close()` is called after the loop.